src/coreclr/tests/src/tracing/regress/GitHub_34839/GitHub_34839.cs

```diff
@@ -0,0 +1,238 @@
+                return false;
+            }
+
+            return true;
+        }
+    }
+
+    class Program
+    {
+        private static MethodInfo m_enableMethod;
+        private static MethodInfo m_disableMethod;
+
+        public static void Enable(TraceConfiguration traceConfig)
+        {
+            m_enableMethod.Invoke(
+                null,
+                new object[]
+                {
+                    traceConfig.ConfigurationObject
+                });
+        }
+
+        public static void Disable()
+        {
+            m_disableMethod.Invoke(
+                null,
+                null);
+        }
+
+        static int Main(string[] args)
+        {
+            TimeSpan profSampleDelay = TimeSpan.FromMilliseconds(1);
+
+            Assembly SPC = typeof(System.Diagnostics.Tracing.EventSource).Assembly;
+            Type eventPipeType = SPC.GetType("System.Diagnostics.Tracing.EventPipe");
+            m_enableMethod = eventPipeType.GetMethod("Enable", BindingFlags.NonPublic | BindingFlags.Static);
+            m_disableMethod = eventPipeType.GetMethod("Disable", BindingFlags.NonPublic | BindingFlags.Static);
+
+            // Setup the configuration values.
+            uint circularBufferMB = 1024; // 1 GB
+            uint level = 5; // Verbose
+
+            const uint configurationCnt = 5;
+
+            // Create 5 instances of EventPipeConfiguration.
+            TraceConfiguration[] configs = new TraceConfiguration[configurationCnt];
+
+            for (int i = 0; i < configurationCnt; i++)
+            {
+                string outputFile = $"default{i}.nettrace";
+                configs[i] = new TraceConfiguration(outputFile, circularBufferMB);
+
+                // Setup the provider values.
+                // Public provider.
+                string providerName = "Microsoft-Windows-DotNETRuntime";
+                UInt64 keywords = 0x4c14fccbd;
+
+                // Enable the provider.
+                configs[i].EnableProvider(providerName, keywords, level);
+
+                // Private provider.
+                providerName = "Microsoft-Windows-DotNETRuntimePrivate";
+                keywords = 0x4002000b;
+
+                // Enable the provider.
+                configs[i].EnableProvider(providerName, keywords, level);
+
+                // Sample profiler.
+                providerName = "Microsoft-DotNETCore-SampleProfiler";
+                keywords = 0x0;
+
+                // Enable the provider.
+                configs[i].EnableProvider(providerName, keywords, level);
+
+                // Set the sampling rate.
+                configs[i].SetSamplingRate(profSampleDelay);
+            }
+
+            // Enable tracing many times
+            for (int i = 0; i < configurationCnt; i++)
+            {
+                Enable(configs[i]);
+            }
+
+            // Sleep for 5 second to let some events to be written 
+            Thread.Sleep(1000);
```

Comment:
Comment doesn't match code.  Also, does the sleep need to be 1s or could it be something shorter, like 500ms?